internal/db_impl/sqlite3/read_ops.go

```diff
@@ -0,0 +1,599 @@
+}
+
+func (r readOps) GetMailboxMessageCount(ctx context.Context, mboxID imap.InternalMailboxID) (int, error) {
+	query := fmt.Sprintf("SELECT COUNT(*) FROM %v WHERE `%v` = ?",
+		v0.UIDsTableName,
+		v0.UIDsFieldMailboxID,
+	)
+
+	return MapQueryRow[int](ctx, r.qw, query, mboxID)
+}
+
+func (r readOps) GetMailboxMessageCountWithRemoteID(ctx context.Context, mboxID imap.MailboxID) (int, error) {
+	internalID, err := r.GetMailboxIDFromRemoteID(ctx, mboxID)
+	if err != nil {
+		return 0, err
+	}
+
+	query := fmt.Sprintf("SELECT COUNT(*) FROM %v WHERE `%v` = ?",
+		v0.UIDsTableName,
+		v0.UIDsFieldMailboxID,
+	)
+
+	return MapQueryRow[int](ctx, r.qw, query, internalID)
+}
+
+func (r readOps) GetMailboxFlags(ctx context.Context, mboxID imap.InternalMailboxID) (imap.FlagSet, error) {
+	query := fmt.Sprintf("SELECT `%v` FROM %v WHERE `%v` = ?",
+		v0.MailboxFlagsFieldValue,
+		v0.MailboxFlagsTableName,
+		v0.MailboxFlagsFieldMailboxID,
+	)
+
+	flags, err := MapQueryRows[string](ctx, r.qw, query, mboxID)
+	if err != nil {
+		return imap.FlagSet{}, err
+	}
+
+	return imap.NewFlagSetFromSlice(flags), nil
+}
+
+func (r readOps) GetMailboxPermanentFlags(ctx context.Context, mboxID imap.InternalMailboxID) (imap.FlagSet, error) {
+	query := fmt.Sprintf("SELECT `%v` FROM %v WHERE `%v` = ?",
+		v0.MailboxPermFlagsFieldValue,
+		v0.MailboxPermFlagsTableName,
+		v0.MailboxPermFlagsFieldMailboxID,
+	)
+
+	flags, err := MapQueryRows[string](ctx, r.qw, query, mboxID)
+	if err != nil {
+		return imap.FlagSet{}, err
+	}
+
+	return imap.NewFlagSetFromSlice(flags), nil
+}
+
+func (r readOps) GetMailboxAttributes(ctx context.Context, mboxID imap.InternalMailboxID) (imap.FlagSet, error) {
+	query := fmt.Sprintf("SELECT `%v` FROM %v WHERE `%v` = ?",
+		v0.MailboxAttrsFieldValue,
+		v0.MailboxAttrsTableName,
+		v0.MailboxAttrsFieldMailboxID,
+	)
+
+	flags, err := MapQueryRows[string](ctx, r.qw, query, mboxID)
+	if err != nil {
+		return imap.FlagSet{}, err
+	}
+
+	return imap.NewFlagSetFromSlice(flags), nil
+}
+
+func (r readOps) GetMailboxUID(ctx context.Context, mboxID imap.InternalMailboxID) (imap.UID, error) {
+	query := fmt.Sprintf("SELECT `%v` FROM %v WHERE `%v` = ? ",
+		v0.MailboxesFieldUIDNext,
+		v0.MailboxesTableName,
+		v0.MailboxesFieldID,
+	)
+
+	return MapQueryRow[imap.UID](ctx, r.qw, query, mboxID)
+}
+
+func (r readOps) GetMailboxMessageCountAndUID(ctx context.Context, mboxID imap.InternalMailboxID) (int, imap.UID, error) {
+	count, err := r.GetMailboxMessageCount(ctx, mboxID)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	uid, err := r.GetMailboxUID(ctx, mboxID)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	return count, uid, nil
+}
+
+func (r readOps) GetMailboxMessageForNewSnapshot(ctx context.Context, mboxID imap.InternalMailboxID) ([]db.SnapshotMessageResult, error) {
+	query := "SELECT `t1`.`remote_id`, GROUP_CONCAT(`t2`.`value`) AS `flags`, `ui_ds`.`recent`, `ui_ds`.`deleted`, `ui_ds`.`uid`, `ui_ds`.`uid_message` FROM `ui_ds`" +
```

Comment:
Yes and No. This is a very complicated query. I can add the constants there, but this was taken verbatim from the output of ent's request. This will be completely gone in the follow up changes I have planned.